src/app/options/options.component.spec.ts

```diff
@@ -18,10 +21,51 @@ describe('OptionsComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(OptionsComponent);
     component = fixture.componentInstance;
+    component.formats = [ "test1", "test2 [Read-only]", "test3 [Write-only]" ];
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should filter out write-only results from input drop-down list', async () => {
+    await fixture.whenStable();
+    const input = fixture.debugElement.query(By.css('.inputAutocomplete')).nativeElement;
+
+    expect(input.innerText).toEqual(" test1 test2 [Read-only]");
+  });
+
+  it('should filter results for input drowdown list based on input', async () => {
+    await fixture.whenStable();
+    const inputInput = fixture.debugElement.query(By.css('.inputInput')).nativeElement;
+
+    inputInput.value = "1";
```

Comment:
Instead of querying for the css selector and then setting the value directly, can you incorporate the use of harnesses?  https://material.angular.io/guide/using-component-harnesses#using-a-harness-to-interact-with-an-angular-material-component